spec/pagination/view-entity-pagination.module.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable max-classes-per-file */
+import { Controller, forwardRef, Module } from '@nestjs/common';
+import { InjectRepository, TypeOrmModule } from '@nestjs/typeorm';
+import { Type } from 'class-transformer';
+import { Repository, ViewColumn, ViewEntity } from 'typeorm';
+
+import { Crud, CrudController, CrudOptions, CrudService, Method, PaginationType } from '../../src';
+import { BaseEntity } from '../base/base.entity';
+import { BaseService } from '../base/base.service';
+import { TestHelper } from '../test.helper';
+
+@ViewEntity('base_view', {
+    expression: `
+        SELECT DISTINCT id, name, type % 2 as category FROM base
+    `,
+})
+export class BaseView {
+    @ViewColumn()
+    @Type(() => String)
+    id: string;
+
+    @ViewColumn()
+    @Type(() => String)
+    name: string;
+
+    @ViewColumn()
+    @Type(() => Number)
+    category: number;
+}
```

Comment:
It's needed to verify whether pagination works correctly even for View Entities without primary keys.